### PR TITLE
[DC-598] Start on refactoring import data

### PR DESCRIPTION
**Pull request by s-rubenstein**

### CODE REVIEW

File: src/pages/ImportData.js
Comment (s-rubenstein) [inline src/pages/ImportData.js:412]: I think it is much more understandable with these chunks pulled out, but would like a second opinion. I think they could potentially be moved to their own files as well, which might reduce the amount in this file and make it easier to parse through the complexity.

File: src/pages/ImportData.js
Comment (s-rubenstein) [inline src/pages/ImportData.js:394]: I left Utils.DEFAULT here, because I think it is hard to name, and it doesn't hurt understanding of what options are in the switch to have it here.

File: src/pages/ImportData.js
Comment (s-rubenstein) [inline src/pages/ImportData.js:214]: I left Utils.DEFAULT here because its hard to name well, and it doesn't hurt understanding of what options are in the switch to leave as this.

File: src/pages/ImportData.js
Comment (s-rubenstein) [inline src/pages/ImportData.js]: I made a few renames because I thought they were clearer (and more searchable in the code.)

I didn't rename at this level because they are stored in the query params, and thus I wouldn't want to accidentally break anything by renaming them.

File: src/pages/ImportData.js
Comment (fboulnois) [inline src/pages/ImportData.js]: This function seems to be doing a lot and requires many arguments, but I'm not sure how it could be broken apart into smaller more concise chunks.

**Review by fboulnois — APPROVED** [review]
Overall looks reasonable, I'm wondering if `ImportDataDestination` could further be refactored.

File: src/pages/ImportData.js
Comment (s-rubenstein) [reply src/pages/ImportData.js]: Great question. I think it will always have a lot of arguments, just because we need to pass that data down from the parent, but I think we could potentially abstract the individual cases into their own global level components. I wasn't sure whether the value there would outweigh needing to plumb through the props to all of them, as opposed to leaving them as methods within this component. (If we do pull them out globally, I would probably also pull out the default case here for consistency)

File: src/pages/ImportData.js
Comment (pshapiro4broad) [inline src/pages/ImportData.js:84]: This code wasn't changed as part of this but personally I don't like UIs that use forms like `(s)`. I think the UI should either count the number and then pluralize appropriately, or always use plural and be wrong the in the case where there's only one entry.

File: src/pages/ImportData.js
Comment (pshapiro4broad) [inline src/pages/ImportData.js:80]: When we migrate to typescript, will these change to using a fixed list of typed arguments? Using a property list is nice in the case where some arguments are omitted, but it doesn't work as well for typed arguments and isn't as useful if no arguments are omitted.

File: src/pages/ImportData.js
Comment (pshapiro4broad) [inline src/pages/ImportData.js]: The only use of `isDataset` in this method is to gate the display of `noteMessage`. If the caller passed in `noteMessage` instead then I think `isDataset` could be removed, and this could be written as:
```suggestion
    noteMessage && div({ style: { marginTop: '0.5rem', lineHeight: '1.5' } }, [noteMessage]),
```

which to me makes it clearer that the test is correct

**Review by pshapiro4broad — COMMENTED** [review]
Still looking, here are my comments so far

**Review by pshapiro4broad — APPROVED** [review]
Looks good, this seems like an improvement to me.

File: src/pages/ImportData.js
Comment (s-rubenstein) [reply src/pages/ImportData.js:80]: Yes these will be a list of typed arguments eventually. (Potentially a typed object if we don't want to worry about argument order)

File: src/pages/ImportData.js
Comment (s-rubenstein) [reply src/pages/ImportData.js:84]: I am going to defer this to the future, because I wanted to hold off on making functional changes to the UI in this PR, if that's alright?

**Pull request merged**
